sycl/source/detail/platform_impl.cpp

```diff
@@ -282,6 +286,73 @@ static void filterAllowList(vector_class<RT::PiDevice> &PiDevices,
   PiDevices.resize(InsertIDx);
 }
 
+// Filter out the devices that are not compatible with SYCL_DEVICE_FILTER.
+// All three entries (backend:device_type:device_num) are optional.
+// The missing entries are constructed using '*', which means 'any' | 'all'.
+// This function matches devices in the order of backend, device_type, and
+// device_num.
+static void filterDeviceFilter(vector_class<RT::PiDevice> &PiDevices,
+                               const plugin &Plugin) {
+  device_filter_list *FilterList = SYCLConfig<SYCL_DEVICE_FILTER>::get();
+  if (!FilterList)
+    return;
+
+  backend Backend = Plugin.getBackend();
+  int InsertIDx = 0;
+  int DeviceNum = 0;
+  for (RT::PiDevice Device : PiDevices) {
+    // Since device type is not determined yet, we use string search for
+    // a known word in the device name information.
```

Comment:
Could you please elaborate more on what it means? Why can't device type be queried here?
Matching device name doesn't look like a robust solution.

Reply:
This function is the place that checks if incompatible devices are filtered even before we create sycl::device.
At this point, we only have piDevices, which is lower level object.
Once sycl::device is created the device_type information is determined.
This is the same approach as done in filterAllowList() at line 220. No device info is available yet.

Reply:
piDeviceGetInfo( PI_DEVICE_INFO_TYPE) can be used in order to determine type of a low level object. Probably it makes sense to add a specialization of get_device_info for internal use.

Reply:
I meant adding `get_device_info` specialization for device type.
Please, check using device type(using piDeviceGetInfo( PI_DEVICE_INFO_TYPE) here directly or thru `get_device_info` specialization) instead of checking device names.

Reply:
Got it.
I get RT::PiDeviceType using piDeviceGetInfo and map them to sycl device_type.